server/chat-commands/room-sections.ts

```diff
@@ -0,0 +1,94 @@
+'gaming' | 'languages' | 'entertainment' | 'lifehobbies' | 'onsitegames';
+
+export const sections: RoomSection[] = [
+	'none', 'nonpublic', 'officialrooms', 'officialtiers', 'communityprojects', 'gaming', 'languages', 'entertainment', 'lifehobbies', 'onsitegames',
+];
+
+export const sectionNames: {[k in RoomSection]: string} = {
+	none: 'none',
+	nonpublic: 'Non-public',
+	officialrooms: 'Official rooms',
+	officialtiers: 'Official tiers',
+	communityprojects: 'Community projects',
+	gaming: 'Gaming',
+	languages: 'Languages',
+	entertainment: 'Entertainment',
+	lifehobbies: 'Life & hobbies',
+	onsitegames: 'On-site games',
+};
+
+export const commands: ChatCommands = {
+	setroomsection: 'roomsection',
+	roomsection(target, room, user) {
+		room = this.requireRoom();
+		if (!target) {
+			if (!this.runBroadcast()) return;
+			this.sendReplyBox(Utils.html`This room is ${sectionNames[room.settings.section] !== 'none' ? `in the ${sectionNames[room.settings.section]} section` : `not in a section`}.`);
+			return;
+		}
+		this.checkCan('gdeclare');
+		const section = room.adjustSection(target);
+		this.sendReply(`The room section is now: ${sectionNames[section]}`);
+
+		this.privateModAction(`(${user.name} changed the room section to: "${sectionNames[section]}".)`);
+		this.modlog('ROOMSECTION', null, `to "${sectionNames[section]}"`);
+	},
+	roomsectionhelp: [
+		`/roomsection [section] - Sets the room this is used in to the specified [section]. Requires: &`,
+		`Valid sections: ${sections.join(', ')}`,
+	],
+
+	desectionleader: 'sectionleader',
+	sectionleader(target, room, user, connection, cmd) {
+		this.checkCan('gdeclare');
+		if (!target) return this.parse(`/help sectionleader`);
+
+		target = this.splitTarget(target, true);
+		const targetUser = this.targetUser;
+		const userid = toID(this.targetUsername);
+		const name = targetUser ? targetUser.name : this.targetUsername;
+		const demoting = cmd === 'desectionleader';
+		if (targetUser?.sectionLeader && !demoting) {
+			throw new Chat.ErrorMessage(`${name} is already a Section Leader.`);
+		} else if (targetUser && !targetUser.sectionLeader && demoting) {
+			throw new Chat.ErrorMessage(`${name} is not a Section Leader.`);
+		}
+		const staff = Rooms.get('staff');
+		if (!demoting) {
+			Users.sectionLeaders.add(userid, name);
+			this.addGlobalModAction(`${name} was appointed Section Leader by ${user.name}.`);
+			this.globalModlog(`SECTION LEADER`, userid);
+			if (staff && !staff.auth.has(userid)) this.parse(`/msgroom staff,/roomvoice ${userid}`);
+			if (targetUser) {
+				targetUser.popup(`You were appointed Section Leader by ${user.name}.`);
+				if (!targetUser.inRooms.has('staff')) this.parse(`/msgroom staff,/invite ${name}`);
+				if (staff && !targetUser.inRooms.has(staff.roomid)) {
+					this.parse(`/msgroom staff,/invite ${name}`);
+				}
+			}
+		} else {
+			Users.sectionLeaders.delete(userid);
```

Comment:
Why isn't this in `Users.globalAuth`?

Reply:
Or, I suppose since this would override the two keys, why not attach it to the room object directly, in room.auth (or the like)?

Reply:
because you can't have `Users.globalAuth.get('kris')` and then have it return `Section Leader` and `+`